File: development/palindrome_word/palindrome_word.cpp
#include<iostream>
#include<string>

bool isPalindrome(std::string word)
{
    int left = 0;
    int right = word.length() - 1;

    while(left < right)
    {
        if(word[left] != word[right]) return false;

        left++;
        right--;
    }

    return true;
}

int main()
{
    std::string word1 = "11011";
    std::string word2 = "code";
    std::cout << "the word: " << word1 << " is palindrome: " << isPalindrome(word1) << std::endl;
    std::cout << "the word: " << word2 << " is palindrome: " << isPalindrome(word2) << std::endl;
    return 0;
}

File: development/palindrome_word/PalindromeWord.scala
object Main extends App {
    def isPalindrome(word: String): Boolean = {
        var left = 0
        var right = word.length - 1

        while (left < right) {
            if(word(left) != word(right)) {
                return false
            }

            left += 1
            right -= 1
        }

        true
    }

    var word1 = "noon"
    var word2 = "artificial"
    println("the word: " + word1 + " is palindrome: " + isPalindrome(word1))
    println("the word: " + word2 + " is palindrome: " + isPalindrome(word2))
}